google/resource_compute_instance.go

```diff
@@ -718,6 +726,10 @@ func expandComputeInstance(project string, d *schema.ResourceData, config *Confi
 		return nil, fmt.Errorf("Error creating guest accelerators: %s", err)
 	}
 
+	if d.Get("status") != "RUNNING" {
```

Comment:
I'm sorry but I did not understand your comment after the first sentence :cry: Could you add further explanations please? Thanks :smile: 
To explain this line, the goal here is to check that only `RUNNING` value can be set at instance creation (there is no way to create a stopped instance). I put that check here because the `expandComputeInstance` function is only used in `resourceComputeInstanceCreate`.

Reply:
https://github.com/terraform-providers/terraform-provider-google/blob/master/google/resource_compute_instance.go#L1407 calls it as well. You've got a few options:

1. Move this check out of expand
1. Stop using expand at 1407 (all of the data pulled from it is available in the `ResourceData` object `d`)
1. Implement a CustomizeDiff for this check, returning an error if it's create time and the value of the field is not `RUNNING`.

#3 is the hardest to implement but also the best user experience- it'll cause an error at plan time instead of apply time.

Reply:
Ok, got it! I did not understand because when I made this PR, `expandComputeInstance` was not used anywhere but in `resourceComputeInstanceCreate`, I did not notice that changed. I've implemented the 3rd solution (CustomizeDiff).